account_move_propagate_ref/models/account_move.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2021 Camptocamp (https://www.camptocamp.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def copy_data(self, default=None):
+        """Add `ref` to copied fields
+
+        The same feature could've been achieved by setting `copy=True` in
+        the field's definition, but it would've made the field copiable
+        everywhere.
+        Instead, we just need to copy it when reversing a move and creating a
+        new one from it, which is done by calling :meth:`copy_data`.
+        """
+        self.ensure_one()
+        if self._context.get("propagate_ref"):
```

Comment:
```suggestion
        if self.env.context.get("propagate_ref"):
```
if prefer accessing the context through the public properties.
even though it's unlikely, specially here, private properties are not guaranteed to stay the same

not blocking of course 👍🏻